### PR TITLE
Fix SendMsgs, delete Send, and introduce GetMsgResult

### PR DESCRIPTION
**Pull request by siburu**

### CODE REVIEW

File: core/send.go [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+	"time"
+
+	retry "github.com/avast/retry-go"
+	"github.com/cosmos/cosmos-sdk/types"
+)
+
+// SendCheckMsgs is an utility function that executes `Chain::SendMsgs` and checks the execution results of all the messages.
+func SendCheckMsgs(chain Chain, msgs []types.Msg) bool {
+	if _, err := chain.SendMsgs(msgs); err != nil {
+		GetChainLogger(chain).Error("failed to send msgs", err)
```
[the remark this answers is not in the document]
Comment (siburu) [reply core/send.go:14]: No. I will add them.

File: chains/tendermint/chain.go [hunk 2]
```diff
@@ -166,16 +169,31 @@ func (c *Chain) RegisterMsgEventListener(listener core.MsgEventListener) {
...
 	if err != nil {
 		return nil, err
+	} else if res.Code != 0 {
+		// CheckTx failed
+		return nil, fmt.Errorf("CheckTx failed: %v", errors.ABCIError(res.Codespace, res.Code, res.RawLog))
 	}
-	if res.Code == 0 && c.msgEventListener != nil {
+
+	// wait for tx being committed
+	if resTx, err := c.waitForCommit(res.TxHash); err != nil {
```
[the remark this answers is not in the document]
Comment (siburu) [reply chains/tendermint/chain.go:187]: Yes!

File: chains/tendermint/chain.go [hunk 3]
```diff
@@ -323,25 +384,66 @@ func CalculateGas(
...
-		return false
+func (c *Chain) GetMsgResult(id core.MsgID) (core.MsgResult, error) {
+	msgID, ok := id.(*MsgID)
+	if !ok {
+		return nil, fmt.Errorf("unexpected message id type: %T", id)
+	}
+
+	// find tx
+	resTx, err := c.waitForCommit(msgID.txHash)
+	if err != nil {
```
[the remark this answers is not in the document]
Comment (siburu) [reply chains/tendermint/chain.go:422]: No. Even if `resTx.TxResult.IsErr() == true`, `GetMsgResult` returns `(msgResult, nil)`.
At that case, `msgResult.Status` returns `(false, failureReason)`.

**Review by bluele — APPROVED** [review]
Awesome, thanks! LGTM👍

**Pull request merged**